packages/lib/src/vis/tiles/TiledHeatmap.tsx

```diff
@@ -28,18 +28,19 @@ function TiledHeatmap(props: Props) {
   const { numLayers } = api;
 
   const dataPerPixel = Math.max(1, xDataPerPixel, yDataPerPixel);
-  const quality = 1 - clamp(qualityFactor, 0, 1);
-  const levelOfDetail = Math.min(
-    Math.floor(Math.log2(dataPerPixel) + quality),
+  const roundingOffset = 1 - clamp(qualityFactor, 0, 1);
+  const subsamplingLevel = Math.min(
+    Math.floor(Math.log2(dataPerPixel) + roundingOffset),
     numLayers - 1
   );
+  const currentLayerIndex = numLayers - 1 - subsamplingLevel;
```

Comment:
Looks like reversing the order of the layers simplified a couple things, that's good! You could maybe just add another getter for `numLayers - 1`, like `lastLayerIndex` or `topLayerIndex` or `maxLayerIndex` ... something like that? Cause it seems you need it in a few places.

Reply:
Right!

Reply:
I used `baseLayerIndex` and renamed `imageSize` getter to `baseLayerSize` for consistency.